utils/litemap/src/map.rs

```diff
@@ -71,6 +72,20 @@ impl<'a, K, V> LiteMap<K, V, &'a [(K, V)]> {
     }
 }
 
+impl<K: ?Sized, V: ?Sized, S> LiteMap<K, V, S>
+where
+    S: StoreConstEmpty<K, V>,
+{
+    /// Create a new empty [`LiteMap`]
+    pub const fn new_empty() -> Self {
```

Comment:
Suggestion: We don't need both `new` (currently only defined for `Vec`) and `new_empty`. For `Vec` they're the same, so just call this one `new`.